internal/storage/blob/store.go

```diff
@@ -295,8 +329,14 @@ func (s *Store) pollForUpdates(ctx context.Context) {
 			s.log.Info("Stopped polling for updates")
 			return
 		case <-ticker.C:
-			if err := s.updateIndex(ctx); err != nil {
+			err := s.updateIndex(ctx)
+			buildErr := &indexBuildError{}
+			if err != nil {
 				s.log.Errorw("Failed to check for updates", "error", err)
```

Comment:
Move this to the `else` block where we check whether the error is an index build error.